src/redux/services/compare/transferResource.ts

```diff
@@ -101,21 +101,18 @@ async function deployResourceToCluster(
 async function extractResourceToLocal(
   source: K8sResource,
   target: K8sResource | undefined,
-  state: RootState
+  dispatch: AppDispatch
 ): Promise<K8sResource> {
-  const resourceMap = state.main.resourceMap;
-  const fileMap = state.main.fileMap;
-
   if (target) {
-    const result = {...target};
-    performResourceContentUpdate(result, source.content, fileMap, resourceMap);
+    const result = structuredClone(target);
+    result.text = source.text;
+    await dispatch(updateResource({resourceId: target.id, text: source.text}));
```

Comment:
This should already fix the bug where I mutate frozen state. A bulk approach is still in the back of my mind but with lower priority.

Reply:
I didn't know you can `await` dispatches, interesting!